### PR TITLE
Fixing IIS task issues

### PR DESCRIPTION
**Pull request by mvvsubbu**

### CODE REVIEW

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppDeploy/MsDeployOnTargetMachines.ps1 [hunk 1]
```diff
@@ -184,6 +189,6 @@ function Execute-Main
     Write-Verbose "AdditionalArguments = $AdditionalArguments"
 
     $overRideParams = Compute-MsDeploy-SetParams -websiteName $websiteName -overRideParams $overRideParams
-    Deploy-Website -webDeployPkg $WebDeployPackage -webDeployParamFile $WebDeployParamFile -overRiderParams $OverRideParams -websiteName $websiteName -excludeFilesFromAppData $excludeFilesFromAppData -removeAdditionalFiles $removeAdditionalFiles -takeAppOffline $takeAppOffline -additionalArguments $AdditionalArguments
```
Comment (pavanadepu2) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppDeploy/MsDeployOnTargetMachines.ps1]: Why it gets removed?

Comment (pavanadepu2) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppDeploy/MsDeployOnTargetMachines.ps1]: Debugging of the task is becoming critical due to missing of verbose logs in few methods. Can you accommodate them.

Comment (mvvsubbu) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppDeploy/MsDeployOnTargetMachines.ps1]: Where do you want me to add?

**Review by pavanadepu2 — APPROVED** [review]
Signed Off

**Comment by shamnigurat** [comment]
pavanadepu2,
mvvsubbu has fixed the issue in the absolutely wrong way.
It's unacceptable because without fixing this issue as I mentioned ([#115](https://github.com/Microsoft/vsts-rm-extensions/issues/115)) deployment is not working at all.
Just look at the parameters of the "WinRM - IIS Web App Deployment" task.
There is the parameter named "Website Name". Just tell me: what is this parameter for?
The information's hint tells: "Provide the name of the website **for deploying the Web project to**."
But this parameter is not used while deploying to IIS so I can't deploy as hint tells me!
Further more, just imagine the case: I have zipped web-package (msdeploy) and want to deploy it to IIS. All I know is the name of the web site to deploy - how can I deploy in this case (I don't know the physical path of the web site and I don't need it because I would be able to deploy on differnet environments with the same site's name!)? The "Override Parameters" doesn't work for this purpose even if I set 'name="IIS Web Application Name",value="...."' - msdeploy just tries to unpack the content of the web-package to the physical path the package was created from.
Wihtout fixing my [issue](https://github.com/Microsoft/vsts-rm-extensions/issues/115) the task "WimRM - IIS Web App Deployment" becomes almost useless.

**Pull request merged**